packages/enhanced/src/lib/sharing/utils.ts

```diff
@@ -330,35 +328,39 @@ function normalizeVersion(versionDesc: string): string {
 
 export { normalizeVersion };
 
+/** @typedef {{ data: Record<string, any>, path: string }} DescriptionFile */
+
+interface DescriptionFile {
+  data: Record<string, any>;
+  path: string;
+}
+
 /**
  *
  * @param {InputFileSystem} fs file system
  * @param {string} directory directory to start looking into
  * @param {string[]} descriptionFiles possible description filenames
- * @param {function((Error | null)=, {data?: object, path?: string}=, (checkedDescriptionFilePaths: string[])=): void} callback callback
- * @param {function({data: Record<string, any>, path: string}=): boolean} satisfiesDescriptionFileData file data compliance check
+ * @param {function((Error | null)=, {data?: DescriptionFile, path?: string}=, (checkedDescriptionFilePaths: string[])=): void} callback callback
+ * @param {function({data: DescriptionFile}=): boolean} satisfiesDescriptionFileData file data compliance check
  */
-const getDescriptionFile = (
+function getDescriptionFile(
   fs: InputFileSystem,
   directory: string,
   descriptionFiles: string[],
   callback: (
     err: Error | null,
-    data?: { data: object; path: string },
+    data?: DescriptionFile,
     checkedDescriptionFilePaths?: string[],
   ) => void,
-  satisfiesDescriptionFileData?: (data: {
-    data: Record<string, any>;
-    path: string;
-  }) => boolean,
-) => {
+  satisfiesDescriptionFileData?: (data?: DescriptionFile) => boolean,
+) {
   let i = 0;
 
-  // use a function property to store the list of visited files inside the recursive lookup
-  // instead of the public getDescriptionFile property
-  const satisfiesDescriptionFileDataInternal:
-    | (typeof satisfiesDescriptionFileData & { checkedFilePaths?: Set<string> })
-    | undefined = satisfiesDescriptionFileData;
+  // Create an object to hold the function and the checkedFilePaths
+  const satisfiesDescriptionFileDataInternal = {
+    check: satisfiesDescriptionFileData,
+    checkedFilePaths: new Set<string>(),
```

Comment:
@ScriptedAlchemy but this will create a new Set instance for each getDescriptionFile call and only the filepaths from the last instance will be passed to the callback?

Reply:
Okay i added it as a default to the function, then pass the same set recursively down.